### PR TITLE
SREP-1199 Implement RHOBS Probes API integration and Probe Reconciliation

### PR DESCRIPTION
**Pull request by dustman9000**
Implement complete probe reconciliation workflow that retrieves synthetic probe configurations from RHOBS Probes API and creates corresponding probe.monitoring.rhobs Custom Resources with URL validation and status tracking.

Core Features:
- API client for RHOBS Probes API with label selector support
- URL validation logic to prevent false positive alerts
- Custom Resource generation for probe.monitoring.rhobs
- Bidirectional status updates (created/failed) via PATCH API calls
- Configurable polling with graceful shutdown

API Integration:
- GET /api/metrics/v1/{tenant}/probes with label filtering
- PATCH /api/metrics/v1/{tenant}/probes/{id} for status updates
- Support for tenant-specific API calls
- Comprehensive error handling and retry logic

Configuration:
- API base URL, tenant, and label selector configuration
- Kubernetes namespace targeting
- Multi-source config (files, env vars, CLI flags)
- Example configuration with sensible defaults

URL Validation:
- HTTP/HTTPS scheme validation
- Reachability testing via HEAD requests
- Server error handling (5xx=fail, 4xx=ok)
- Prevents creation of non-functional monitoring

Testing:
- Comprehensive unit tests (67.7% coverage)
- Integration tests with mock API servers
- Error handling and edge case coverage
- Race condition detection support

Documentation:
- Complete README with architecture and usage
- API reference and troubleshooting guide
- Test coverage summary and development guide
- Enhanced Makefile with categorized targets

[SREP-1199](https://issues.redhat.com/browse/SREP-1199)

### CODE REVIEW

**Review by jimdaga — COMMENTED** [review]
Please see comments

File: internal/api/client.go [hunk 1]
```diff
@@ -1,3 +1,129 @@
...
+		Tenant:  tenant,
+		HTTPClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}
+}
+
+// GetProbes retrieves probes from the API with optional label selectors
+func (c *Client) GetProbes(labelSelector string) ([]Probe, error) {
+	endpoint := fmt.Sprintf("/api/metrics/v1/%s/probes", c.Tenant)
```
[the remark this answers is not in the document]
Comment (dustman9000) [reply internal/api/client.go]: Added support to talk to api directly with/without JWT token support.

File: internal/k8s/probe.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+			"namespace": pm.namespace,
+			"labels": map[string]string{
+				"rhobs.monitoring/probe-id":           probe.ID,
+				"rhobs.monitoring/cluster-id":         clusterID,
+				"rhobs.monitoring/management-cluster": managementClusterID,
+				"rhobs.monitoring/managed-by":         "rhobs-synthetics-agent",
+			},
+		},
+		Spec: ProbeSpec{
+			Interval:  "30s",
```
[the remark this answers is not in the document]
Comment (dustman9000) [reply internal/k8s/probe.go]: Good callout. Added a config for this.

File: internal/k8s/probe.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	// Extract cluster information from labels
+	clusterID := probe.Labels["cluster_id"]
+	managementClusterID := probe.Labels["management_cluster_id"]
+	
+	if clusterID == "" {
+		clusterID = "unknown"
+	}
+	if managementClusterID == "" {
+		managementClusterID = "unknown"
+	}
```
[the remark this answers is not in the document]
Comment (dustman9000) [reply internal/k8s/probe.go]: updated
